SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -209,6 +215,25 @@ QueryConstraint InterestFactory::CreateAlwaysInterestedConstraint()
 	return AlwaysInterestedConstraint;
 }
 
+
+QueryConstraint InterestFactory::CreateSingletonConstraint()
+{
+	QueryConstraint SingletonConstraint;
+
+	Worker_ComponentId SingletonComponentIds[] = {
+		SpatialConstants::SINGLETON_COMPONENT_ID,
+		SpatialConstants::SINGLETON_MANAGER_COMPONENT_ID };
```

Comment:
nit: The singleton manager is only required on servers, the client early-outs all it's logic. Might be worth switching on if this is being attached to the RPC client end point, or the position component.

Reply:
I was curious about whether it was needed or not, but it's in the streaming queries for the client which is why I included it. Do you think it's worth making that change now, or as part of the larger refactor to apply interest to a single entity per worker?

Reply:
Yep, can refactor when working in the larger interest updates.